packages/flame/lib/src/geometry/line_segment.dart

```diff
@@ -11,6 +11,8 @@ class LineSegment {
 
   factory LineSegment.zero() => LineSegment(Vector2.zero(), Vector2.zero());
 
+  Vector2 get midpoint => (from + to) / 2;
```

Comment:
```suggestion
  Vector2 get midpoint => (from + to)..scale(0.5);
```
To not create an unnecessary object